[teesha902] ip

teesha902 opened the pull request:
# 🐷 **PiggyPlanner**  

> “The secret of getting ahead is getting started.” – *Mark Twain*

PiggyPlanner is your personal task management assistant that helps you **organize**, **prioritize**, and **conquer** your daily tasks without breaking a sweat. It's:

- 📝 **Text-based** for simplicity
- 🚀 **Lightning-fast** and easy to use
- 🎯 Focused on **productivity** and **clarity**

## 🚀 **Getting Started**

1. Clone the repository from [[PiggyPlanner Repo](https://github.com/teesha902/ip)].
2. Navigate to the project directory and build the JAR:
   ```bash
   ./gradlew shadowJar
   ```
3. You’ll find the JAR file in the `build/libs` folder:
   - **`PiggyPlanner.jar`**

4. To run the application:
   ```bash
   java -jar PiggyPlanner.jar
   ```

## ✨ **Features**  
- 📝 Add **To-Dos**, **Deadlines**, and **Events** effortlessly.
- 📅 View your agenda for specific dates with:
  ```bash
  agenda for 2/12/2023
  ```
- ✅ Mark tasks as complete:
  ```bash
  mark 1
  ```
- ❌ Delete tasks:
  ```bash
  delete 3
  ```
- [ ] **Task Reminders** *(coming soon!)*  
- [ ] **Recurring Tasks** *(in development!)*

## 🔧 **For Developers**  
If you're learning Java, **PiggyPlanner** is also a great way to sharpen your skills. Here's the entry point of the app:

```java
public class PiggyPlanner {
    public static void main(String[] args) {
        try {
            new PiggyPlanner().run();
        } catch (PiggyException e) {
            System.out.println("An error occurred while starting the program: " + e.getMessage());
        }
    }
}
```
---

Ready to let **PiggyPlanner** simplify your life? 🐷✨  
Give it a try, and feel free to contribute or leave feedback! 🚀

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

teesha09 committed:
Completed Level 0: Changed bot name, added greeting and goodbye

teesha09 committed:
Level 1 completed: Added echo and 'bye' command functionality

teesha09 committed:
Level 2 completed: Made taskList to store all tasks, able add user input into taskList

teesha09 committed:
Level 3 completed: Added Task class, added mark/unmark task functionality

teesha09 committed:
Task subclasses created (ToDo, Event, Deadline)

teesha09 committed:
Level 4 completed: Task class and subclasses made, main program logic adjusted to accomodate

teesha09 committed:
A-TextUiTesting completed: Edited the runtest.sh file, wrote comprehensive test cases in input.txt and their corresponding expected outputs in EXPECTED.txt

teesha09 committed:
Reorganised into packages (commands, tasks) to facilitate clean exception handling

teesha09 committed:
Level 5 completed: Error handling and extensive test cases written

teesha09 committed:
Level 6 completed: Added 'delete' command and tests for it

teesha09 committed:
A-Enum level completed: Enum created for different command types

teesha09 committed:
Delete Duke.java file

teesha09 committed:
Add *.class files to .gitignore

teesha09 committed:
Remove all wildcard imports in files that had them

teesha09 committed:
Update task constructors to include new overloaded constructor with isDone boolean

teesha09 committed:
Add storage class to handle output/saving to disk

teesha09 committed:
Complete Level-7: Add functionality for saving list and handle corrupted files from previous saves

teesha09 committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Complete Level-7: Add functionality for saving list and handle corrupted files from previous saves
  Add storage class to handle output/saving to disk
  Update task constructors to include new overloaded constructor with isDone boolean

# Conflicts:
#	src/main/java/PiggyPlanner.java

teesha09 committed:
Update Deadline.java to use java.time.LocalDateTime instead of storing time information in strings

teesha09 committed:
pdate Event.java to use java.time.LocalDateTime instead of storing time information in strings

teesha09 committed:
Adjust AddTask.java to format new tasks in correct format and throw error pointing users to correct date/time format for input

teesha09 committed:
Update Storage.java to ensure task list is saved and updated in correct format

teesha09 committed:
Add functionality for new command: "DayPlan.java" for listing deadlines and events for specified day

teesha09 committed:
Add "today" and "tomorrow" into valid time inputs for DayPlan.java

teesha09 committed:
Ensure event start time/date is before end time/date

teesha09 committed:
Complete Level 8: Change date and time information from strings to java.time.LocalDate or java.time.LocalDateTime objects

teesha09 committed:
Complete A-OOP: Made program use more OOP by using Ui class to handle messages to print to screen and a TaskList class to have a TaskList object. Rest of the program already OOP based.

teesha09 committed:
Merge branch 'more-oop'

* more-oop:
  Complete A-OOP: Made program use more OOP by using Ui class to handle messages to print to screen and a TaskList class to have a TaskList object. Rest of the program already OOP based.

teesha09 committed:
Commit gitignore

teesha09 committed:
Merge branch 'add-gradle-support'

* add-gradle-support:
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

teesha09 committed:
Complete A-Gradle: Set up gradle for project

teesha09 committed:
Create dummy test to play with J-Unit and gradle

teesha09 committed:
Add J-Unit tests for Deadline.java and Event.java

teesha09 committed:
Complete J-Unit: Add tests for all commands' public methods

teesha09 committed:
Complete A-Javadoc: Add header comments to all non-private classes/methods, and non-trivial private methods from in all program files

teesha09 committed:
Add "find {keyword}" command functionality

teesha09 committed:
Add functionality for searching multiple keywords at a time

teesha09 committed:
Complete Level-9: Add "find" command functionality and corresponding Javadocs and J-Unit tests

teesha09 committed:
Make minor change to be able to push level-8 branch for system detection

teesha09 committed:
Make another minor change to play with level-8 branch

teesha09 committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Make another minor change to play with level-8 branch

teesha09 committed:
Make minor changes for javadoc branch for system detection

teesha09 committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Make minor changes for javadoc branch for system detection

teesha09 committed:
Complete A-CodingStandard. Make changes for program to align better with given coding standards

teesha09 committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Complete A-CodingStandard. Make changes for program to align better with given coding standards

teesha09 committed:
Minor change for system detection of branch-level-9

teesha09 committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Minor change for system detection of branch-level-9

teesha09 committed:
Add checkstyle

teesha09 committed:
Add GUI basic elements (window, layout, colors)

teesha09 committed:
Add functionality to keyboard 'Enter' key to work as 'Send' button

teesha09 committed:
Add back original PiggyPlanner logic as the responses in the GUI

teesha09 committed:
Merge branch 'branch-A-Checkstyle'

* branch-A-Checkstyle:
  Add checkstyle

teesha09 committed:
Fix text formatting on GUI window

teesha09 committed:
Add icons for PiggyPlanner and user

teesha09 committed:
Allow window resizing properly

teesha09 committed:
Complete Level 10: GUI

teesha09 committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Complete Level 10: GUI
  Allow window resizing properly
  Add icons for PiggyPlanner and user
  Fix text formatting on GUI window
  Add back original PiggyPlanner logic as the responses in the GUI
  Add functionality to keyboard 'Enter' key to work as 'Send' button
  Add GUI basic elements (window, layout, colors)

teesha09 committed:
Complete A-Varargs: Add use of varargs in "Find" command so that 1 or more keywords can be search for at once

teesha09 committed:
Merge branch 'branch-Varargs'

* branch-Varargs:
  Complete A-Varargs: Add use of varargs in "Find" command so that 1 or more keywords can be search for at once

brvndonkoh commented on src/main/java/commands/Find.java:
Is it possible to rename this method to something clearer? Perhaps executeFind() to better convey its purpose?

brvndonkoh commented on src/main/java/commands/ListCommand.java:
Is it possible to rename execute() to executeList() instead? Since many of your classes have an execute() function, renaming this one would improve clarity and make it easier to distinguish, not just for you but also for the grader.

lilduckling replied:
Maybe you can make one abstract `Command` class with `execute` method, it would be more closely follow OOP

brvndonkoh commented on src/main/java/commands/Mark.java:
Is it possible to rename execute() to executeMark()? Since multiple classes have an execute() function, renaming this one would improve clarity and make it easier to differentiate.

brvndonkoh commented on src/main/java/commands/Unmark.java:
Is it possible to rename execute() to executeUnmark()? Since multiple classes have an execute() function, renaming this one would improve clarity and make it easier to differentiate, especially for maintainability and grading.

brvndonkoh commented on src/main/java/piggyplanner/PiggyPlanner.java:
There are many execute() methods in the main code. If they are renamed as suggested in the comments, it would make the code much clearer and easier to follow.

brvndonkoh commented on src/test/java/commands/DeleteTaskTest.java:
The execute() method in this file is well-named and clear. No changes needed!

brvndonkoh reviewed (commented):
Overall, the naming is well done, except for the few I pointed out. Keep up the great work, Teesha!

lilduckling commented on src/main/java/commands/AddTask.java:
good job following coding standard!

teesha09 committed:
Fix Find and FindTest to be consistent with other command class formats

teesha09 committed:
Fix checkstyle configuration issue when running gradlew build

teesha09 committed:
Fix checkstyle issues and evolved test outpyts that came up when rebuilding app

teesha09 committed:
Complete A-Assertions

teesha09 committed:
Make small changes

teesha09 committed:
Merge branch 'branch-A-Assertions'

* branch-A-Assertions:
  Complete A-Assertions
  Fix checkstyle issues and evolved test outpyts that came up when rebuilding app
  Fix checkstyle configuration issue when running gradlew build
  Fix Find and FindTest to be consistent with other command class formats

# Conflicts:
#	src/main/java/commands/Find.java
#	src/test/java/commands/FindTest.java

teesha09 committed:
Improve code quality of Find.java by breaking main method into smaller methods and better following SLAP and SRP principles

teesha09 committed:
Include asserts in Find.java and update its Find.java tests

teesha09 committed:
Update DayPlan.java to make methods shorter and adhere better to code quality standard

teesha09 committed:
Refactor Storage.java to split up large main function into smaller functions better suited for code quality standard

teesha09 committed:
Refactor Find command to use streams for keyword matching

The Find command previously used nested loops to match tasks against user-provided keywords. This resulted in imperative-style code with reduced readability.

The code is now refactored to use Java Streams for better readability and conciseness. Streams simplify the checking of whether any keyword is present in a task’s name.

Using streams improves the clarity of the matching logic by expressing it declaratively, making the intent clearer.

teesha902 committed:
Merge pull request #2 from teesha902/branch-A-CodeQuality

Improve code quality

teesha09 committed:
Merge branch 'master' into branch-A-Streams

* master:
  Refactor Storage.java to split up large main function into smaller functions better suited for code quality standard
  Update DayPlan.java to make methods shorter and adhere better to code quality standard
  Include asserts in Find.java and update its Find.java tests
  Improve code quality of Find.java by breaking main method into smaller methods and better following SLAP and SRP principles

# Conflicts:
#	src/main/java/commands/Find.java

teesha902 committed:
Merge pull request #1 from teesha902/branch-A-Streams

Implement streams in Find command

teesha09 committed:
Add HelpCommand to display available commands and formats

The PiggyPlanner application currently lacks an in-built guide to inform users of available commands and their respective formats. This can cause frustration for new users unfamiliar with the input structure.

To improve user experience, a HelpCommand has been added. When the user enters the "help" command, the program displays a comprehensive list of commands with their expected formats. This allows users to quickly learn how to interact with the PiggyPlanner.

teesha09 committed:
Merge branch 'C-Help'

* C-Help:
  Add HelpCommand to display available commands and formats

teesha09 committed:
Merge branch 'master' of github.com:teesha902/ip

* 'master' of github.com:teesha902/ip:
  Refactor Find command to use streams for keyword matching

teesha09 committed:
Add .yml for CI

teesha09 committed:
Change color of user/chatbot text boxes

teesha09 committed:
Add timestamps under each dialogBox

teesha09 committed:
Prevent multiple Jar files from being created

teesha09 committed:
Merge branch 'branch-A-BetterGui'

* branch-A-BetterGui:
  Add timestamps under each dialogBox
  Change color of user/chatbot text boxes

teesha09 committed:
Add error checking for adding duplicate tasks

teesha09 committed:
Merge branch 'branch-A-MoreErrorHandling'

* branch-A-MoreErrorHandling:
  Add error checking for adding duplicate tasks

teesha09 committed:
Add JUnit Tests for TaskList.java

teesha09 committed:
Add JUnit tests for PiggyPlanner.java

teesha09 committed:
Merge branch 'branch-A-MoreTesting'

* branch-A-MoreTesting:
  Add JUnit tests for PiggyPlanner.java
  Add JUnit Tests for TaskList.java

teesha09 committed:
Update README page

teesha09 committed:
Merge branch 'branch-A-UserGuide'

* branch-A-UserGuide:
  Update README page

teesha09 committed:
Add link to user guide when help command called

teesha09 committed:
Change format slightly for link

teesha902 committed:
Update README.md

teesha902 committed:
Update README.md

teesha09 committed:
Add missing configuration for windows/linux for javaFX in gradle file

teesha09 committed:
Merge commit '0af82b04936d2ed0f993eae9d8d6edc29463df37'

* commit '0af82b04936d2ed0f993eae9d8d6edc29463df37':
  Update README.md
  Update README.md

teesha09 committed:
Fix starting window size (issue on Windows only)
